core/services/relay/evm/method_binding.go

```diff
@@ -28,9 +31,22 @@ func (m *methodBinding) SetCodec(codec commontypes.RemoteCodec) {
 	m.codec = codec
 }
 
-func (m *methodBinding) Bind(_ context.Context, binding commontypes.BoundContract) error {
-	m.address = common.HexToAddress(binding.Address)
+func (m *methodBinding) Bind(ctx context.Context, binding commontypes.BoundContract) error {
```

Comment:
Can you add a test for this?  I'm not sure if it's an interface test to make sure no one can make the mistake again or if it's EVM only.  It feels like it can be for both methods and events in the generic one.

Reply:
I was thinking the events one runs a check on the database, and it doesn't actually call the contract anyway. The method does an eth_call and that was the underlying issue.

I can add a test for it though to hold the function expectation.

Reply:
I added a test for the error case.